### DISCUSSION TITLE
Setting min log level for G004 and other logging rules

### DISCUSSION THREAD

**Discussion by Apakottur** [astral-sh/ruff#13304] [Ideas]
In a project I'm working on, it would be useful if the rule G004 would apply to
```python
logging.info(f"This is a prod log which should not be an f-string: {value}")
```

But not to:
```python
logging.debug(f"This is a debug log that I want to allow f-strings for: {value}")
```

Is there a way to apply the logging rules only to info+ logs (and not debug logs), or maybe there should be a way to do that?

**Comment by MichaReiser** [comment]
Hi @Apakottur 

That's interesting. Could you tell me a bit more about your motivation for wanting to flag `info` log calls but not `debug`?

**Reply by Apakottur** [reply]
We have a different configuration for our logger depenending on the environment it's being run at.
One of the differences in configuration applies to debug logs:
1. In local development environment, debug logs are printed (together with some color formatting etc)
2. In all other environments debug logs are dropped and never printed.

We want very strict checks for our non debug logs, to make sure that we follow all the best practices of logging, but don't care so much about that with debug logs, which are only used for local development.
It's of course best to follow practices for all logs, but we don't find fixing our debug logs as important and would love to be able to automatically ignore them from these checks.

**Reply by MichaReiser** [reply]
Thanks for explaining. I do understand the reasoning behind it, but I'm not convinced it adds the complexity of adding an option. I'm mainly concerned that an option communicates to users that it's fine to not follow the practice for some log levels which may not be true for your project (it is in your case)

**Reply by Apakottur** [reply]
Makes perfect sense, and I think that in our case we should also strive to follow the rules for all the logs.
I was just looking for a way to ignore the debug logs, at least for a while, so that we can start by fixing the non debug logs that matter, and then later fix the debug logs (which we have hundreds of).

**Reply by MichaReiser** [reply]
Yeah, I can understand that. Unfortunately, the rules don't have an auto-fix because fixing all the debug calls would be easy.  I haven't considered whether a fix is feasible or why it is missing. I don't know if you would find it fun to contribute a fix.
